newIDE/app/src/MainFrame/index.js

```diff
@@ -1930,6 +1986,19 @@ const MainFrame = (props: Props) => {
     });
   };
 
+  const onChangeProjectName = async (newName: string): Promise<void> => {
+    if (!currentProject || !currentFileMetadata) return;
+    const storageProviderOperations = getStorageProviderOperations();
+    if (storageProviderOperations.onChangeProjectProperty) {
+      await storageProviderOperations.onChangeProjectProperty(
+        currentProject,
+        currentFileMetadata,
+        { name: newName }
+      );
+      updateWindowTitle();
```

Comment:
This is a bit "imperative", any chance this could be done automatically in an effect by checking the storage provider and the `currentProject.getName()` directly? :)

Reply:
I'm not sure what to watch in the effect dependencies. `getStorageProvider` does not change, neither does `currentProject`. If the `currentProject` name is modified, the MainFrame component does not see a change in its props.